src/lib.rs

```diff
@@ -70,10 +70,72 @@ pub static README_PATHS: &[&str; 5] = &[
 
 pub static LICENSE_PATHS: &[&str; 3] = &["LICENSE", "LICENSE.md", "COPYING"];
 
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PackageName {
+    pub namespace: String,
+    pub name: String,
+}
+
+impl PackageName {
+    /// Parses the package name from a `namespace/name` format
+    pub fn parse(s: &str) -> Result<Self, &'static str> {
+        if !s.contains('/') {
+            return Err("no / in package name");
+        }
+        let mut split = s.split('/');
+        let name = split.next().ok_or("no name in package name")?.to_string();
+        let namespace = split
+            .next()
+            .ok_or("no namespace in package name")?
+            .to_string();
+        Ok(Self { name, namespace })
+    }
+
+    /// Returns the `namespace/name` package name
+    pub fn to_string(&self) -> String {
+        format!("{}", self)
+    }
+}
+
+impl std::str::FromStr for PackageName {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::parse(s)
+    }
+}
+
+impl fmt::Display for PackageName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}/{}", self.namespace, self.name)
+    }
+}
+
+mod serde_package_name {
+    use super::PackageName;
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(package: &PackageName, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        s.serialize_str(&package.to_string())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<PackageName, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let buf = String::deserialize(deserializer)?;
+        PackageName::parse(&buf).map_err(serde::de::Error::custom)
+    }
+}
```

Comment:
Can these be moved to `impl serde::Serialize for PackageName` and `impl<'de> serde::Deserialize<'de> for PackageName` blocks? Otherwise it'll be difficult for downstream crates to store a `PackageName` as a field.

Reply:
ok done